File: components/autofill/core/browser/data_model/loyalty_card.cc
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/autofill/core/browser/data_model/loyalty_card.h"

namespace autofill {

LoyaltyCard::LoyaltyCard(std::string loyalty_card_id,
                         std::string merchant_name,
                         std::string program_name,
                         std::string program_logo,
                         std::string loyalty_card_number)
    : loyalty_card_id(std::move(loyalty_card_id)),
      merchant_name(std::move(merchant_name)),
      program_name(std::move(program_name)),
      program_logo(std::move(program_logo)),
      loyalty_card_number(std::move(loyalty_card_number)) {}

LoyaltyCard::LoyaltyCard(const LoyaltyCard&) = default;
LoyaltyCard::LoyaltyCard(LoyaltyCard&&) = default;

LoyaltyCard::~LoyaltyCard() = default;

}  // namespace autofill

File: components/autofill/android/java/src/org/chromium/components/autofill/LoyaltyCard.java
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.autofill;

import org.jni_zero.CalledByNative;
import org.jni_zero.JNINamespace;
import org.jni_zero.JniType;

import org.chromium.build.annotations.NullMarked;
import org.chromium.url.GURL;

import java.util.List;

/** Represents a loyalty card coming from Google Wallet. */
@JNINamespace("autofill")
@NullMarked
public class LoyaltyCard {
    private final String mLoyaltyCardId;
    private final String mMerchantName;
    private final String mProgramName;
    private final GURL mProgramLogo;
    private final String mLoyaltyCardNumber;
    private final List<GURL> mMerchantDomains;

    @CalledByNative
    public LoyaltyCard(
            @JniType("std::string") String loyaltyCardId,
            @JniType("std::string") String merchantName,
            @JniType("std::string") String programName,
            @JniType("GURL") GURL programLogo,
            @JniType("std::string") String loyaltyCardNumber,
            @JniType("std::vector<GURL>") List<GURL> merchantDomains) {
        mLoyaltyCardId = loyaltyCardId;
        mMerchantName = merchantName;
        mProgramName = programName;
        mProgramLogo = programLogo;
        mLoyaltyCardNumber = loyaltyCardNumber;
        mMerchantDomains = merchantDomains;
    }

    @CalledByNative
    public @JniType("std::string") String getLoyaltyCardId() {
        return mLoyaltyCardId;
    }

    @CalledByNative
    public @JniType("std::string") String getMerchantName() {
        return mMerchantName;
    }

    @CalledByNative
    public @JniType("std::string") String getProgramName() {
        return mProgramName;
    }

    @CalledByNative
    public @JniType("GURL") GURL getProgramLogo() {
        return mProgramLogo;
    }

    @CalledByNative
    public @JniType("std::string") String getLoyaltyCardNumber() {
        return mLoyaltyCardNumber;
    }

    @CalledByNative
    public @JniType("std::vector<GURL>") List<GURL> getMerchantDomains() {
        return mMerchantDomains;
    }
}
